### PR TITLE
Favorite & Share Permissions

### PR DESCRIPTION
**Pull request by kobergj**
Adds new permissions as described in https://github.com/owncloud/ocis/issues/7497 

Needs reva PR to enforce these permissions.

### CODE REVIEW

File: services/settings/pkg/store/defaults/permissions.go
Comment (JammingBen) [inline services/settings/pkg/store/defaults/permissions.go:1]: I don't know much about the exact code here, but could it be that the `Share.Write` permission is missing here? 🤔 At least I don't see it.

Comment (kobergj) [reply services/settings/pkg/store/defaults/permissions.go:1]: It is here: https://github.com/owncloud/ocis/pull/7700/files#diff-6dad214221d8e9023e7d9a8a99a9d9446553d5c8c97be4d877274c49777e9c0aR89-R106 

We agreed on `Shares.Write` I think.

Comment (JammingBen) [reply services/settings/pkg/store/defaults/permissions.go:1]: Ah, I was looking for `Share.Write`. Then there is a typo in the changelog item 😄 The AC's in https://github.com/owncloud/ocis/issues/7497 also say `Share`, but tbh I don't mind - I just need to know.

Comment (kobergj) [reply services/settings/pkg/store/defaults/permissions.go:1]: Actually the ticket says both :smile: so I was throwing a die to choose what to use...

Comment (kobergj) [reply services/settings/pkg/store/defaults/permissions.go:1]: I fixed it in the changelog :+1:

**Review by JammingBen — APPROVED** [review]

**Review by 2403905 — CHANGES_REQUESTED** [review]

File: services/settings/pkg/store/defaults/permissions.go [hunk 1]
```diff
@@ -356,15 +413,15 @@ func SpaceAbilityPermission(c settingsmsg.Permission_Constraint) *settingsmsg.Se
 	}
 }
 
-// WritePublicLinkPermission is the permission to write public links
-func WritePublicLinkPermission(c settingsmsg.Permission_Constraint) *settingsmsg.Setting {
+// WriteFavoritesPermission is the permission to mark/unmark files as favorites
+func WriteFavoritesPermission(c settingsmsg.Permission_Constraint) *settingsmsg.Setting {
 	return &settingsmsg.Setting{
-		Id:          "11516bbd-7157-49e1-b6ac-d00c820f980b",
-		Name:        "PublicLink.Write",
```
Comment (rhafer) [inline services/settings/pkg/store/defaults/permissions.go]: This removes the `PublicLink.Write` permission. Do we really want that?

**Review by rhafer — CHANGES_REQUESTED** [review]

File: services/settings/pkg/store/defaults/permissions.go (hunk 1 above)
Comment (kobergj) [reply services/settings/pkg/store/defaults/permissions.go]: No we don't. But isn't it still here: https://github.com/owncloud/ocis/pull/7700/files#diff-6dad214221d8e9023e7d9a8a99a9d9446553d5c8c97be4d877274c49777e9c0aR70-R87 ?

File: services/graph/pkg/service/v0/drives.go [hunk 2]
```diff
@@ -586,13 +586,13 @@ func (g Graph) ListStorageSpacesWithFilters(ctx context.Context, filters []*stor
...
 	_, err = s.GetPermissionByID(ctx, &settingssvc.GetPermissionByIDRequest{
-		PermissionId: settingsServiceExt.ListAllSpacesPermissionID,
+		PermissionId: settingsServiceExt.ListSpacesPermission(0).Id,
 	})
 
 	permissions := make(map[string]struct{}, 1)
 	// No error means the user has the permission
 	if err == nil {
-		permissions[settingsServiceExt.ListAllSpacesPermissionName] = struct{}{}
+		permissions[settingsServiceExt.ListSpacesPermission(0).Id] = struct{}{}
```
[the remark this answers is not in the document]
Comment (kobergj) [reply services/graph/pkg/service/v0/drives.go]: Fixed :+1:

File: services/settings/pkg/store/defaults/permissions.go (hunk 1 above)
Comment (rhafer) [reply services/settings/pkg/store/defaults/permissions.go]: Ah, sorry. Overlooked that.

**Review by 2403905 — APPROVED** [review]

**Review by butonic — APPROVED** [review]

**Pull request merged**